cg/meta/demultiplex/housekeeper_storage_functions.py

```diff
@@ -43,6 +46,40 @@ def store_flow_cell_data_in_housekeeper(
     add_demux_logs_to_housekeeper(
         flow_cell=flow_cell, hk_api=hk_api, flow_cell_run_dir=flow_cell_run_dir
     )
+    add_undetermined_fastq_files_to_housekeeper(flow_cell=flow_cell, hk_api=hk_api)
+
+
+def add_undetermined_fastq_files_to_housekeeper(
+    flow_cell: FlowCellDirectoryData, hk_api: HousekeeperAPI
+) -> None:
+    """Add undetermined fastq files for non-pooled samples in Housekeeper."""
+    sample_sheet: SampleSheet = get_sample_sheet(flow_cell)
+    non_pooled_samples: List[FlowCellSample] = sample_sheet.get_non_pooled_samples()
+
+    for sample in non_pooled_samples:
+        undetermined_fastqs: List[Path] = get_undetermined_fastqs(
+            lane=sample.lane, flow_cell_path=flow_cell.path
+        )
+
+        for fastq_path in undetermined_fastqs:
+            add_fastq_file_to_housekeeper(
+                sample_id=sample.sample_id,
+                flow_cell_id=flow_cell.id,
+                fastq_path=fastq_path,
+                hk_api=hk_api,
+            )
+
+
+def add_fastq_file_to_housekeeper(
```

Comment:
Do we not have a function for this when adding the normal FASTQ files?

Reply:
That function also does a check against the sequencing metrics table whether the fastq file should be tracked. Probably a SRP violation, the check should be extracted. Can be refactored in a separate PR to limit the scope of this one

Reply:
Makes sense 👍